### PR TITLE
Remove unused code.

### PR DESCRIPTION
**Pull request by Shylock-Hg**
## What type of PR is this?
- [ ] bug
- [ ] feature
- [ ] enhancement

#### Issue(s) number: 
Close https://github.com/vesoft-inc/nebula/issues/3840
## Checklist:
Tests:
- [ ] Unit test(positive and negative cases)
- [ ] Function test
- [ ] Performance test
- [ ] N/A

Affects:
- [ ] Documentation affected (Please add the label if documentation needs to be modified.)
- [ ] Incompatibility (If it breaks the compatibility, please describe it and add the label.）
- [ ] If it's needed to cherry-pick (If cherry-pick to some branches is required, please label the destination version(s).)
- [ ] Performance impacted: Consumes more CPU/Memory

## Release notes:

Please confirm whether to be reflected in release notes and how to describe:
> ex. Fixed the bug .....

### CODE REVIEW

**Review by Aiee — APPROVED** [review]

**Review by yixinglu — APPROVED** [review]

File: src/common/time/parser/DatetimeReader.h [hunk 1]
```diff
@@ -16,22 +16,12 @@ namespace time {
 
 class DatetimeReader {
  public:
+  DatetimeReader();
```
Comment (jackwener) [inline src/common/time/parser/DatetimeReader.h:19]: It may be better add `= default`

**Comment by jackwener** [comment]
LGTM

File: src/common/time/parser/DatetimeReader.h (hunk 1 above)
Comment (Shylock-Hg) [reply src/common/time/parser/DatetimeReader.h:19]: Can't be default.

**Review by dutor — APPROVED** [review]
LGTM.

**Pull request merged**